internal/elasticsearch/transform/transform_test.go

```diff
@@ -0,0 +1,387 @@
+func testAccResourceTransformWithPivotCreate(name string) string {
+	return fmt.Sprintf(`
+provider "elasticstack" {
+  elasticsearch {}
+}
+
+resource "elasticstack_elasticsearch_transform" "test_pivot" {
+  name = "%s"
+	description = "test description"
+
+	source {
+		indices = ["source_index_for_transform"]
+	}
+
+	destination {
+		index = "dest_index_for_transform"
+	}
+
+	pivot = jsonencode({
+    "group_by": {
+      "customer_id": {
+        "terms": {
+          "field": "customer_id",
+          "missing_bucket": true
+        }
+      }
+    },
+    "aggregations": {
+      "max_price": {
+        "max": {
+          "field": "taxful_total_price"
+        }
+      }
+    }
+  })
+
+	sync {
+    time {
+      field = "order_date"
+      delay = "20s"
+    }
+  }
+
+	max_page_search_size = 2000
+  frequency = "5m"
+	enabled = false
+
+	defer_validation = true
+	timeout = "1m"
+}
+	`, name)
+}
+
+// update the existing transform, add another source index and start it (enabled = true)
+// validations are now unavoidable (at start), so make sure to create the indices _before_ updating the transform
+// the tf script below uses implicit dependency, but `depends_on` is also an option
+func testAccResourceTransformWithPivotUpdate(name string) string {
+	return fmt.Sprintf(`
+provider "elasticstack" {
+  elasticsearch {}
+}
+
+resource "elasticstack_elasticsearch_index" "test_source_index_1" {
+  name = "source_index_for_transform"
+
+  alias {
+    name = "test_alias_1"
+  }
+
+  mappings = jsonencode({
+    properties = {
+      field1 = { type = "text" }
+    }
+  })
+
+	deletion_protection = false
+	wait_for_active_shards = "all"
+	master_timeout = "1m"
+	timeout = "1m"
+}
+
+resource "elasticstack_elasticsearch_index" "test_source_index_2" {
+  name = "additional_index"
+
+  alias {
+    name = "test_alias_2"
+  }
+
+  mappings = jsonencode({
+    properties = {
+      field1 = { type = "text" }
+    }
+  })
+
+	deletion_protection = false
+	wait_for_active_shards = "all"
+	master_timeout = "1m"
+	timeout = "1m"
+}
+
+resource "elasticstack_elasticsearch_transform" "test_pivot" {
+  name = "%s"
+	description = "yet another test description"
+
+	source {
+		indices = [
+			elasticstack_elasticsearch_index.test_source_index_1.name,
+			elasticstack_elasticsearch_index.test_source_index_2.name
+		 ]
+	}
+
+	destination {
+		index = "dest_index_for_transform_v2"
+	}
+
+	pivot = jsonencode({
+    "group_by": {
+      "customer_id": {
+        "terms": {
+          "field": "customer_id",
+          "missing_bucket": true
+        }
+      }
+    },
+    "aggregations": {
+      "max_price": {
+        "max": {
+          "field": "taxful_total_price"
+        }
+      }
+    }
+  })
+
+	sync {
+    time {
+      field = "order_date"
+      delay = "20s"
+    }
+  }
+
+	retention_policy {
+    time {
+      field   = "order_date"
+      max_age = "7d"
+    }
+  }
+
+	max_page_search_size = 1000
```

Comment:
```suggestion
	max_page_search_size = 2000
```

It looks like this acceptance test is failing in 7.13 because we're adjusting this setting. The simplest path forward for this PR might be to stop updating this setting.

Reply:
Yes, after some investigations, it looks like ES versions below 7.16 have some issues when attempting to update some settings on a transform that was never started (probably because some internal metadata was not created, hence the error).  I've modified the acc test as you suggested.